src/braket/pennylane_plugin/translation.py

```diff
@@ -560,93 +563,88 @@ def translate_result_type(  # noqa: C901
             return DensityMatrix(targets)
         raise NotImplementedError(f"Unsupported return type: {return_type}")
 
-    if isinstance(observable, (Hamiltonian, qml.Hamiltonian)):
-        if return_type is ObservableReturnTypes.Expectation:
-            return tuple(
-                Expectation(_translate_observable(term), term.wires) for term in observable.ops
-            )
-        raise NotImplementedError(f"Return type {return_type} unsupported for Hamiltonian")
-
     if observable is None:
         if return_type is ObservableReturnTypes.Counts:
-            return tuple(Sample(observables.Z(), target) for target in targets or measurement.wires)
+            return tuple(Sample(observables.Z(target)) for target in targets or measurement.wires)
         raise NotImplementedError(f"Unsupported return type: {return_type}")
 
+    observable = _flatten_observable(observable)
+
+    if isinstance(observable, qml.ops.LinearCombination):
+        if return_type is ObservableReturnTypes.Expectation:
+            return tuple(Expectation(_translate_observable(op)) for op in observable.terms()[1])
+        raise NotImplementedError(f"Return type {return_type} unsupported for Hamiltonian")
+
     braket_observable = _translate_observable(observable)
     if return_type is ObservableReturnTypes.Expectation:
-        return Expectation(braket_observable, targets)
+        return Expectation(braket_observable)
     elif return_type is ObservableReturnTypes.Variance:
-        return Variance(braket_observable, targets)
+        return Variance(braket_observable)
     elif return_type in (ObservableReturnTypes.Sample, ObservableReturnTypes.Counts):
-        return Sample(braket_observable, targets)
+        return Sample(braket_observable)
     else:
         raise NotImplementedError(f"Unsupported return type: {return_type}")
 
 
+def _flatten_observable(observable):
+    if isinstance(observable, (qml.ops.Hamiltonian, qml.ops.CompositeOp, qml.ops.SProd)):
+        simplified = qml.ops.LinearCombination(*observable.terms()).simplify()
+        coeffs, _ = simplified.terms()
+        if len(coeffs) > 1 or coeffs[0] != 1:
```

Comment:
Why do we skip this if `coeffs == 1`?

Reply:
In this case, the observable is a flat tensor product, which we support directly.